Feature/advertisments list

tavisca-mrathore opened the pull request:
@achirametavisca @tavisca-smohite @tavisca-rnayyar

Commit:
added service to get advertisment list data from mock api and minor changes

Commit:
ui made responsive for and details service added

achirametavisca commented on src/app/services/GetAdvertismentDetails/get-advertisment-details.service.ts:
```diff
@@ -0,0 +1,17 @@
+})
+export class GetAdvertismentDetailsService {
+
+  private AD_DETAILS_REST_API_SERVER = "https://demo3944353.mockable.io/get-advertisment-details";
```
move this to common class

tavisca-mrathore replied:
Done

achirametavisca commented on src/app/app.component.css:
```diff
@@ -0,0 +1,19 @@
+app-advertisments-list {
```
Remove this
home page story covers this.

tavisca-mrathore replied:
Done

achirametavisca commented on src/app/app.component.html:
```diff
@@ -1 +1 @@
-
+<app-advertisments-list></app-advertisments-list>
```
Remove this
home page story covers this.

tavisca-mrathore replied:
Done

achirametavisca commented on src/app/home-module/advertisments-list/advertisments-list.component.html:
```diff
@@ -1 +1,11 @@
-<p>advertisments-list works!</p>
+<div *ngFor="let ad of adsList" class="adv-card">
```
Add routing to this component and test this component separately.

tavisca-mrathore replied:
Done

achirametavisca commented on src/app/services/GetAdvertismentDetails/get-advertisment-details.service.ts:
```diff
@@ -0,0 +1,17 @@
+
+  constructor(private httpClient: HttpClient) { }
+
+  public sendGetRequest() {
```
Give name that describes what this function is doing.

tavisca-mrathore replied:
Done

achirametavisca commented on src/app/services/GetAdvertismentsList/get-advertisments-list.service.ts:
```diff
@@ -0,0 +1,17 @@
+
+  constructor(private httpClient: HttpClient) { }
+
+  public sendGetRequest() {
```
Give name that describes what this function is doing.

tavisca-mrathore replied:
Done

achirametavisca commented on src/app/services/GetAdvertismentDetails/get-advertisment-details.service.ts:
```diff
@@ -0,0 +1,17 @@
+import { Injectable } from '@angular/core';
```
**You don't need two different services**

get-advertisments-list.service.ts 
get-advertisment-details.service.ts

Create one service advertismentsService that will have different methods related to advertisments

tavisca-mrathore replied:
Done

achirametavisca commented on src/app/services/GetAdvertismentDetails/get-advertisment-details.service.ts:
```diff
@@ -0,0 +1,17 @@
+
+  constructor(private httpClient: HttpClient) { }
+
+  public sendGetRequest() {
```
You should mention return type of this function.
This function should return observable<Type>
 ```
/** POST: add a new hero to the database */
addHero (hero: Hero): Observable<Hero> {
  return this.http.post<Hero>(this.heroesUrl, hero, httpOptions)
    .pipe(
      catchError(this.handleError('addHero', hero))
    );
}
```
In component where this function is called, you should handle success, failure cases everytime.
Follow this for all public functions in service
Discuss with me if having any doubt. 
>

tavisca-mrathore replied:
done

achirametavisca commented on src/app/services/GetAdvertismentsList/get-advertisments-list.service.ts:
```diff
@@ -0,0 +1,17 @@
+})
+export class GetAdvertismentsListService {
+
+  private ADS_LIST_REST_API_SERVER = "https://demo4724356.mockable.io/get-advertisments-list";
```
move to common class

tavisca-mrathore replied:
Done

achirametavisca reviewed (changes requested):
Check review comments

tavisca-mrathore committed:
provided support for absolute imports thereby eliminating relative imports, global constants defined in environment.ts

tavisca-mrathore committed:
global constants in a common file, removed ad list component's styling, added descriptive names for service functions, combined list and details services to a common advertisments service

tavisca-mrathore committed:
deleted individual services for advertisment functionalities, added routing for ad list component's testing, removed bootsrap cdn

tavisca-mrathore committed:
deleted individual advertisment services from directory structure

removed solution for imports problem
renamed advertisments-list component to products-list
renamed advertisments service to products

tavisca-mrathore committed:
Merge branch 'develop' of https://github.com/tavisca-rrai/tavisca-exchange-ui into feature/advertisments-list

tavisca-mrathore committed:
handling responses from service now

tavisca-mrathore committed:
ng2-validator added

tavisca-mrathore committed:
handling responses from service now

tavisca-mrathore committed:
Merge branch 'develop' of https://github.com/tavisca-rrai/tavisca-exchange-ui into feature/advertisments-list

tavisca-mrathore committed:
added routing for products list component

tavisca-mrathore committed:
mock for product details added, indentation for routing module

achirametavisca reviewed (approved):
Looks good
